sql/sql_tmp_table.cc

```diff
@@ -2256,6 +2256,9 @@ bool create_innodb_tmp_table(TABLE *table, KEY *keyinfo)
   create_info.row_type= table->s->row_type;
   create_info.options|= HA_LEX_CREATE_TMP_TABLE |
                         HA_LEX_CREATE_INTERNAL_TMP_TABLE;
+
+  table->file->adjust_create_info_for_frm(&create_info);
```

Comment:
`ha_innobase::adjust_create_info_for_frm` is looking at `srv_encrypt_tables`, but for tmp tables we should look at the `temp_tablespace_encrypt`?

Reply:
ah. I missed that `adjust_create_info_for_frm` is changed

Reply:
No, the conclusion from yesterday's discussion with Robert & Laurynas is:
1. innodb_temp_tablespace_encrypt is only meant to enable/disable encryption of temp tablespace. Not for decision to encrypt temp table or not
2. temporary, system and general tablespaces should follow same rules/pattern
3. When tables lacks explicit encryption option, innodb_encrypt_tables is the only option to decide whether the table should be encrypted or not.

Reply:
yes, I remember that. My original comment was about intrinsic tables, but later I noticed that their creation flag is dependant on tablespace encryption flag, which looks right.

Reply:
Right, intrinsic should go by tablespace property.